### ISSUE TITLE
Next button on Frontmatter page

### ISSUE THREAD

**Issue by Alex-Jordan** [rbeezer/mathbook#739]
On the frontmatter page:
http://mathbook.pugetsound.edu/examples/sample-article/html/

when there are no prefaces or other chunked subsections (as in the sample article), the "next" button is grayed out, but I think it could lead to the first thing that follows the frontmatter.

**Comment by rbeezer** [comment]
You mean like how "Reference" (the <backmatter>) has a "Prev"?

I'm not sure where the difference is?

**Comment by Alex-Jordan** [comment]
> I'm not sure where the difference is?

You don't see the symptom? Or you are unsure what the underlying code issue is?

I mean that if you follow that link, the "Next" button is grayed out and can't be clicked. I think it would be reasonable for it to be functional there.

I suspect it is looking for a subsection of the frontmatter, but in this case (and others) there is no subsection. It makes me wonder if a book chapter had no sections, but the chunk level was still 2, if this same issue would happen for that chapter.

**Comment by rbeezer** [comment]
Yes, I certainly can see what was reported.

And I believe it is inconsistent with the Prev button on "backmatter".

And I do not know why without more study of the (complicated) code.

**Comment by rbeezer** [comment]
Would you be willing to test that hypothesis?

>I suspect it is looking for a subsection of the frontmatter, but in
>this case (and others) there is no subsection. It makes me wonder if a
>book chapter had no sections, but the chunk level was still 2, if this
>same issue would happen for that chapter.

**Comment by Alex-Jordan** [comment]
Tested, and the hypothesis is proven false. I don't see the issue in a sectionless chapter, only in a sectionless Frontmatter.

**Comment by rbeezer** [comment]
Thanks.  I might peek at the code today if I get a chance.

**Comment by rbeezer** [comment]
Nice catch.  Now I wonder why nobody noticed, never?  Fixed at  5850453d477cd56a81c11e1ff6c0c66708773f4f

The `frontmatter` has a weird child, `titlepage`, that does not create a new page if chunking that far.  And for an `article`, the only other child, `abstract`, does not either.  But in a `book`, lots of children of `frontmatter` create their own page, like `colophon`.

You would have had no way to know, but the test would be to build a `book/frontmatter` with just a `titlepage` child (which I confirmed).  See the commit for more details on the cause and cure.
